lib/streamaggr/streamaggr.go

```diff
@@ -708,83 +726,101 @@ func (a *aggregator) runFlusher(pushFunc PushFunc, alignFlushToInterval, skipInc
 		}
 	}
 
-	if dedupInterval <= 0 {
-		alignedSleep(interval)
-		t := time.NewTicker(interval)
-		defer t.Stop()
+	flushDeadline := time.Now().Truncate(interval).Add(interval)
+	tickInterval := time.Duration(a.tickInterval) * time.Millisecond
+	alignedSleep(tickInterval)
+
+	var dedupIdx, flushIdx int
 
-		if alignFlushToInterval && skipIncompleteFlush {
-			a.flush(nil, interval, true)
-			ignoreFirstIntervals--
+	t := time.NewTicker(tickInterval)
+	defer t.Stop()
+
+	isSkippedFirstFlush := false
+	for tickerWait(t) {
```

Comment:
As far as I can see, logic of the main flusher loop has changed here. Previously, there were different cases when dedup was enabled and when it wasn't used. For example, `isSkippedFirstFlush` was only used in a code path which was triggered when dedup was enabled.

Is this an expected change? Should we document this behaviour change somewhere(probably in changelog) so that users will expect the behaviour to be changed?